pkg/ibmcsidriver/node_helper_test.go

```diff
@@ -20,18 +20,87 @@
 package ibmcsidriver
 
 import (
+	"fmt"
 	"testing"
 
 	cloudProvider "github.com/IBM/ibmcloud-volume-file-vpc/pkg/ibmcloudprovider"
+	"github.com/stretchr/testify/assert"
 )
 
 func TestProcessMount(t *testing.T) {
+	// test cases
+	testCases := []struct {
+		name        string
+		source      string
+		targetPath  string
+		fsType      string
+		options     []string
+		expectedErr error
+	}{
+		{
+			name:        "success",
+			source:      "/source",
+			targetPath:  "/targetpath",
+			fsType:      "nfs",
+			options:     []string{"ro,sync"},
+			expectedErr: nil,
+		},
+		{
+			name:        "Invalid source path",
+			source:      "./error_mount_source",
+			targetPath:  "/targetpath",
+			fsType:      "nfs",
+			options:     []string{"ro,sync"},
+			expectedErr: fmt.Errorf("fake Mount: source error"),
+		},
+		{
+			name:        "Invalid target path",
+			source:      "./error_mount_target",
+			targetPath:  "fake-volPath",
+			fsType:      "nfs",
+			options:     []string{"ro,sync"},
+			expectedErr: fmt.Errorf("fake Mount: target error"),
+		},
+		{
+			name:        "Make directory fails",
+			source:      "/source",
+			targetPath:  "invalid-volPath-dir",
+			fsType:      "nfs",
+			options:     []string{"ro,sync"},
+			expectedErr: fmt.Errorf("Path Creation failed"),
+		},
+		{
+			name:        "IsLikelyNotMountPoint returns true and nil error",
+			source:      "./error_mount_source",
+			targetPath:  "fake-volPath-1",
+			fsType:      "nfs",
+			options:     []string{"ro,sync"},
+			expectedErr: fmt.Errorf("fake Mount: target error"),
+		},
+		{
+			name:        "Umount Fails",
+			source:      "./error_mount_source",
+			targetPath:  "error_umount",
+			fsType:      "nfs",
+			options:     []string{"ro,sync"},
+			expectedErr: fmt.Errorf("Unmount Failed"),
+		},
+	}
+
 	// Creating test logger
 	logger, teardown := cloudProvider.GetTestLogger(t)
 	defer teardown()
 
-	icDriver := initIBMCSIDriver(t)
-	ops := []string{"a", "b"}
-	response, err := icDriver.ns.processMount(logger, "processMount", "/staging", "/targetpath", "ext4", ops)
-	t.Logf("Response %v, error %v", response, err)
+	// Run test cases
+	for _, tc := range testCases {
+		t.Logf("test case: %s", tc.name)
+		// Setup new driver each time so no interference
+		icDriver := initIBMCSIDriver(t)
+		// Call processMount
+		_, err := icDriver.ns.processMount(logger, "processMount", tc.source, tc.targetPath, tc.fsType, tc.options)
+		if tc.expectedErr != nil {
+			t.Logf("Error code")
+			assert.NotNil(t, err)
```

Comment:
Thats not the good validation, its not validating actual error which is expected for a particular test, you may need to check a specific error and validate as per expectation